packages/@webex/plugin-cc/src/services/core/WebSocket/WebSocketManager.ts

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable */
+import { Signal } from '../Signal';
+import { WebexSDK, SubscribeRequest, HTTP_METHODS, WelcomeResponse } from '../../../types';
+import { SUBSCRIBE_API, WCC_API_GATEWAY } from '../../constants';
+import { SubscribeResponse } from '../../config/types';
+import workerScript from './keepalive.worker';
+
+export class WebSocketManager {
+  readonly onMessage: Signal.WithData<string>;
+  private readonly onMessageSend: Signal.Send<string>;
+  readonly onSocketClose: Signal.Empty;
+  private readonly onSocketCloseSend: Signal.SendEmpty;
+  private websocket: WebSocket;
+  shouldReconnect: boolean;
+  isSocketClosed: boolean;
+  private isWelcomeReceived: boolean;
+  private url: string | null = null;
+  private forceCloseWebSocketOnTimeout: boolean;
+  private isConnectionLost: boolean;
+  private webex: WebexSDK;
+  private welcomePromiseResolve: ((value: WelcomeResponse | PromiseLike<WelcomeResponse>) => void) | null = null;
+
+  private keepaliveWorker: Worker;
+
+  constructor(options: {webex: WebexSDK}) {
+    const {webex} = options;
+    this.webex = webex;
+    const { send, signal } = Signal.create.withData<string>();
+    this.onMessage = signal;
+    this.onMessageSend = send;
+
+    const socketCloseSignal = Signal.create.empty();
+    this.onSocketClose = socketCloseSignal.signal;
+    this.onSocketCloseSend = socketCloseSignal.send;
+    this.shouldReconnect = true;
+    this.websocket = {} as WebSocket;
+    this.isSocketClosed = false;
+    this.isWelcomeReceived = false;
+    this.forceCloseWebSocketOnTimeout = false;
+    this.isConnectionLost = false;
+
+    const workerScriptBlob = new Blob([workerScript], { type: 'application/javascript' });
+    this.keepaliveWorker = new Worker(URL.createObjectURL(workerScriptBlob));
+  }
+
+  async initWebSocket(options: {body: SubscribeRequest}): Promise<WelcomeResponse> {
+    const connectionConfig = options.body;
+    await this.register(connectionConfig);
+    return new Promise((resolve, reject) => {
+      this.welcomePromiseResolve = resolve;
+      this.connect()
+        .catch((error) => {
+          console.error(`[WebSocketStatus] | Error in connecting Websocket`, error);
+          reject(error);
+        });
+    });
+  }
+
+  async reconnect() {
```

Comment:
This method is exposed so that we can reconnect the flow if the network is turned off and gets turned on again. It uses the keep-alive to monitor this and based on the message, it initiates a reconnect logic (found in connection-service.ts, which is taken from agent desktop).